core/src/main/java/io/kestra/core/storages/StorageInterface.java

```diff
@@ -50,9 +54,24 @@ default boolean exists(URI uri) {
     @Retryable(includes = {IOException.class}, excludes = {FileNotFoundException.class})
     Long lastModifiedTime(URI uri) throws IOException;
 
+    @Retryable(includes = {IOException.class}, excludes = {FileNotFoundException.class})
+    Long createdAtTime(URI uri) throws IOException;
+
+    // Not sur what representation you except for permissions ?
+    @Retryable(includes = {IOException.class}, excludes = {FileNotFoundException.class})
+    Long permissions(URI uri) throws IOException;
+
+    // Thought we could use that with no data for empty file creation
     @Retryable(includes = {IOException.class})
     URI put(URI uri, InputStream data) throws IOException;
 
+    // Thought this could work for directory ?
+    @Retryable(includes = {IOException.class})
+    URI put(URI uri) throws IOException;
```

Comment:
I think you should name it createDirectory for better understanding of what it does

Reply:
Yes, it's mandatory